### ISSUE TITLE
Add rotations

### ISSUE THREAD

**Issue by hsgg** [ziotom78/Healpix.jl#74]
[healpy](https://healpy.readthedocs.io/en/latest/index.html) has support for rotations through the [Rotator](https://healpy.readthedocs.io/en/latest/generated/healpy.rotator.Rotator.html) class. Are there any plans to add something similar to Healpix.jl? What would be involved in doing so?

**Comment by ziotom78** [comment]
Rotating maps is a tricky business, as it can be done either in pixel or harmonic space, but each case has its own subtleties… There are several reasons why one wants to perform a rotation of a Healpix map:

-   Convert directions between different coordinate systems (e.g., Ecliptic→Galactic)
-   Convert maps between different coordinate systems
-   Apply arbitrary rotations for other purposes (e.g., Monte Carlo simulations)

The first point should probably be passed to a proper astronomic library (e.g., [Astrolib.jl](https://juliaastro.github.io/AstroLib.jl/stable/) or [Astropy](https://www.astropy.org/) via [PyCall.jl](https://github.com/JuliaPy/PyCall.jl)). The second point is subtle because it can lead to many problems (see [this thread](https://github.com/healpy/healpy/issues/460) on the Healpy mailing list, it was initiated by yours truly) and all the pros and cons of each approach should be carefully documented in the Healpix.jl manual. Finally, general rotations could be implemented using [Rotations.jl](https://github.com/JuliaGeometry/Rotations.jl) or other similar packages.

To sum up, it is surely a useful feature and would be nice to have it integrated in Healpix, but it is quite complex… If anybody volunteers to submit a PR, we can surely have a look at it.

**Comment by pochoi** [comment]
+1

Thank you so much for explaining the difficulties!
Indeed I wish I can do rotation in the harmonic space, i.e. `rotate_alm` in healpix.
I know the challenge is to compute and apply Wigner rotation matrix to alm. I always wish I have Julia codes to do it. If there are any good resources to help me on it, please let me know!

**Comment by hsgg** [comment]
Thanks @ziotom78 ! Given the complexity and subtleties involved, maybe one could start it as a separate package?

For my purposes a relatively simple version could be enough, maybe using [WignerD.jl](https://github.com/jishnub/WignerD.jl).

**Comment by ziotom78** [comment]
Hi @hsgg , I like the idea of a separate package! We might advertise it in the README and the documentation for Healpix,jl.
